src/server/providers/navigraph_dfd/mappers.ts

```diff
@@ -1289,46 +1264,69 @@ export class DFDMappers {
         return `V${navaid.icaoCode}${navaid.airportIdentifier ?? '    '}${navaid.llzIdentifier}`;
     }
 
-    public static decodeIdColumn(id: string): FixInfo | undefined {
-        if (!id) {
-            return undefined;
-        }
+    public static fixTypeToDatabaseIdPrefix(type: FixType): string {
+        if (type === FixType.IlsNavaid) return 'V';
+        return type;
+    }
+
+    public static fixInfoToFix(info: DecodedIdColumn, lat: Latitude, long: Longitude): Fix {
+        return {
+            fixType: info.fixType,
+            databaseId: DFDMappers.mapFixDatabaseId(info),
+            location: { lat, long },
+            icaoCode: info.icaoCode,
+            ident: info.ident,
+        };
+    }
 
+    public static decodeIdColumn(id: string): DecodedIdColumn {
         const [table, mapping] = id.split('|', 2);
 
-        const terminal = table.indexOf('terminal') !== -1 || table === 'tbl_runways' || table === 'tbl_localizers_glideslopes' || table === 'tbl_gls';
-        const area = terminal ? WaypointArea.Terminal : WaypointArea.Enroute;
-        const suppressIcaoCode = table === 'tbl_localizers_glideslopes' || table === 'tbl_gls';
-        const icaoCode = terminal ? mapping.substring(4, 6) : mapping.substring(0, 2);
-        const airportIdent = terminal ? mapping.substring(0, 4) : '    ';
-        const ident = terminal ? mapping.substring(6) : mapping.substring(2);
+        let terminal = table.includes('terminal');
+        let fixType: FixType;
+        if (table.includes('airports')) {
+            fixType = FixType.Airport;
+        } else if (table.includes('gls')) {
+            terminal = true;
+            fixType = FixType.GlsNavaid;
+        } else if (table.includes('localizers_glideslopes')) {
+            terminal = true;
+            fixType = FixType.IlsNavaid;
+        } else if (table.includes('ndbnavaids')) {
+            fixType = FixType.NdbNavaid;
+        } else if (table.includes('runways')) {
+            terminal = true;
+            fixType = FixType.Runway;
+        } else if (table.includes('vhfnavaids')) {
+            fixType = FixType.VhfNavaid;
+        } else if (table.includes('waypoints')) {
+            fixType = FixType.Waypoint;
+        } else {
+            throw new Error(`Unknown table ${table}`);
+        }
```

Comment:
This whole thing really just spreads the logic out and makes it less readable than before... Before it was very easy to see on one line what is considered terminal for example.